lib/droplet_kit/mappings/database_postgres_config_mapping.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module DropletKit
+  class DatabasePostgresConfigMapping
+    include Kartograph::DSL
+
+    kartograph do
+      mapping DatabasePostgresConfig
+      root_key singular: 'config', scopes: %i[read update]
+
+      scoped :read, :update do
+        %i[autovacuum_freeze_max_age
+           autovacuum_max_workers
+           autovacuum_naptime
+           autovacuum_vacuum_threshold
+           autovacuum_analyze_threshold
+           autovacuum_vacuum_scale_factor
+           autovacuum_analyze_scale_factor
+           autovacuum_vacuum_cost_delay
+           autovacuum_vacuum_cost_limit
+           bgwriter_delay
+           bgwriter_flush_after
+           bgwriter_lru_maxpages
+           bgwriter_lru_multiplier
+           deadlock_timeout
+           default_toast_compression
+           idle_in_transaction_session_timeout
+           jit
+           log_autovacuum_min_duration
+           log_error_verbosity
+           log_line_prefix
+           log_min_duration_statement
+           max_files_per_process
+           max_prepared_transactions
+           max_pred_locks_per_transaction
+           max_locks_per_transaction
+           max_stack_depth
+           max_standby_archive_delay
+           max_standby_streaming_delay
+           max_replication_slots
+           max_logical_replication_workers
+           max_parallel_workers
+           max_parallel_workers_per_gather
+           max_worker_processes
+           pg_partman_bgw
+           pg_partman_bgw
+           pg_stat_statements
```

Comment:
These three fields appear as:
```
pg_partman_bgw.role
pg_partman_bgw.interval
pg_stat_statements.track
```
in the JSON, which doesn't play well with how the mappings behave (since instance variables can't have a `.` in them, and each field becomes an instance variable).

We can either delete them for now for the sake of simplicity since this is a large PR, or find a way to make these fields:
```
pg_partman_bgw_role
pg_partman_bgw_interval
pg_stat_statements_track
```

Reply:
Hmm, nice catch. I will do some testing around this and `pgbouncer`/`timescaledb` objects as well

Reply:
@bentranter these dot fields should be working now, and pgbouncer/timescaledb are now nested objects (similar to Go's implementation and docs)